File: tests/csharp/csharp_error_handling/try_catch_finally.cs
// vybe-test: csharp/csharp_error_handling/try_catch_finally
// origin: languages/csharp/tests/csharp/test_csharp_error_handling.rs

void __Check(string got, string want) {
    if (got != want) {
        Console.WriteLine("FAIL: want [" + want + "] got [" + got + "]");
        throw new Exception("assertion failed");
    }
}

try {
    throw new Exception("fail");
} catch (Exception e) {
    __Check(("caught: " + e.Message).ToString(), "caught: fail");
} finally {
    __Check(("cleanup").ToString(), "cleanup");
}

File: tests/vb/vb_error_handling/try_catch_finally.vb
' vybe-test: vb/vb_error_handling/try_catch_finally
' origin: languages/vb/tests/vb/test_vb_error_handling.rs

' Vybe test harness — Visual Basic.
'
' Real VB source alongside harness/go/check.go and harness/js/check.js, the way
' test262's assert.js is JavaScript.
'
' A test's verdict is its EXIT CODE. __Check prints its diagnostic BEFORE
' throwing: an uncaught exception surfaces as `RuntimeError: [object]`, which
' says nothing at all.

Module VybeCheck
    Sub __Check(got As String, want As String)
        If got <> want Then
            Console.WriteLine("FAIL: want [" & want & "] got [" & got & "]")
            Throw New Exception("assertion failed")
        End If
    End Sub
End Module

Module M
    Sub Main()
        Try
            Throw New Exception("fail")
        Catch ex As Exception
            __Check(CStr("caught: " & ex.Message), "caught: fail")
        Finally
            __Check(CStr("cleanup"), "cleanup")
        End Try
    End Sub
End Module
